float32 z_min
float32 z_max
---